Code/GraphMol/SmilesParse/SmilesParse.cpp

```diff
@@ -78,99 +78,44 @@ int smarts_bond_parse(const std::string &inp, Bond *&bond) {
 
   return res;
 }
-int smarts_atom_parse(const std::string &inp, Atom *&atom) {
-  void *scanner;
-  int res = 1;  // initialize with fail code
-
-  TEST_ASSERT(!yysmarts_lex_init(&scanner));
-  try {
-    size_t ltrim = setup_smarts_string(inp, scanner);
-    int start_tok = static_cast<int>(START_ATOM);
-    std::vector<RWMol *> molVect;
-    Bond *lastBond = nullptr;
-    res = yysmarts_parse(inp.c_str() + ltrim, &molVect, atom, lastBond, scanner,
-                         start_tok);
-  } catch (...) {
-    yysmarts_lex_destroy(scanner);
-    throw;
-  }
-  yysmarts_lex_destroy(scanner);
-
-  if (res == 1) {
-    std::stringstream errout;
-    errout << "Failed parsing SMARTS '" << inp << "'";
-    throw SmilesParseException(errout.str());
-  }
+int smarts_bond_parse(const std::string &inp, Bond *&bond) {
+  auto start_tok = static_cast<int>(START_BOND);
+  std::vector<RWMol *> molVect;
+  Atom *atom = nullptr;
+  auto res = smarts_parse_helper(inp, molVect, atom, bond, start_tok);
```

Comment:
Can simplify to

```
return smarts_parse_helper(...)
```

Reply:
I often leave the local variable there because it makes debugging easier. That's not particularly relevant here